datadog_checks_dev/datadog_checks/dev/tooling/commands/meta/snmp/generate_traps_db.py

```diff
@@ -0,0 +1,300 @@
+        code_generator = JsonCodeGen()
+        file_writer = FileWriter(compiled_mibs_sources).setOptions(suffix='.json')
+        mib_compiler = MibCompiler(SmiV1CompatParser(tempdir=''), code_generator, file_writer)
+        mib_compiler.addSources(*getReadersFromUrls(*mib_sources, **dict(fuzzyMatching=True)))
+        mib_compiler.addSearchers(*searchers)
+
+        compiled_mibs, compiled_dependencies_mibs = compile_and_report_status(mib_files, mib_compiler)
+
+        # Move all the parent MIBs that had to be compiled but were not requested in the command to a subfolder.
+        for mib_file_name in compiled_dependencies_mibs:
+            os.replace(
+                os.path.join(compiled_mibs_sources, mib_file_name + '.json'),
+                os.path.join(mibs_sources_dir, mib_file_name + '.json'),
+            )
+
+        # Only generate trap_db with `mib_files` unless explicitly asked. Used to ignore other files that may be
+        # present "compiled_mibs_sources"
+        compiled_mibs = [os.path.join(compiled_mibs_sources, x + '.json') for x in compiled_mibs]
+
+        # Generate the trap database based on the compiled MIBs.
+        trap_db_per_mib = generate_trap_db(compiled_mibs, mibs_sources_dir)
+
+        use_json = output_format == "json"
+        if output_file:
+            # Compact representation, only one file
+            write_compact_trap_db(trap_db_per_mib, output_file, use_json=use_json)
+            echo_success("Wrote trap data to {}".format(os.path.abspath(output_file)))
+        else:
+            # Expanded representation, one file per MIB.
+            write_trap_db_per_mib(trap_db_per_mib, output_dir, use_json=use_json)
+            echo_success("Wrote trap data to {}".format(os.path.abspath(output_dir)))
+
+
+def compile_and_report_status(mib_files, mib_compiler):
+    """
+    Iteratively compile all the mibs into multiple json files.
+    :param mib_files: List of path to mib files to compile
+    :param mib_compiler: A pysnmp compiler object
+    :return: A list of the compiled MIBs and a list of dependant MIBs that had to be compiled as well.
+    """
+    child_compiled_mibs = []
+    all_compiled_mibs = []  # Name of all compiled mibs including the parents recursively
+    with click.progressbar(mib_files, label="Compiling MIBs: ", show_eta=False) as pb:
+        for mib_file in pb:
+            mibs_status = mib_compiler.compile(
+                mib_file, noDeps=False, rebuild=True, dryRun=False, genTexts=True, writeMibs=True, ignoreErrors=False
+            )
+            failed_mibs = {k: v for k, v in mibs_status.items() if v == 'failed'}
+            missing_mibs = [k for k, v in mibs_status.items() if v == 'missing']
+            for mib_name, compilation_status in failed_mibs.items():
+                echo_failure(
+                    '\33[2K\rFailed to compile MIB {}: {}'.format(mib_name, compilation_status.error), indent=False
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Shall we have a constant for `\33[2K\r` ? or an argument to `echo_failure(..., erase_current_line=True)` ?